Sources/Testing/Running/Configuration.swift

```diff
@@ -162,4 +162,33 @@ public struct Configuration: Sendable {
       testFilter = selection.contains
     }
   }
+
+  /// A function that handles filtering test cases.
+  ///
+  /// - Parameters:
+  ///   - testCase: The test case to be filtered.
+  ///   - test: The test which `testCase` is associated with.
+  ///
+  /// - Returns: A Boolean value representing if the test case satisfied the
+  ///   filter.
+  @_spi(ExperimentalParameterizedTesting)
+  public typealias TestCaseFilter = @Sendable (_ testCase: Test.Case, _ test: Test) -> Bool
+
+  /// The filter which should be applied to a specified test's associated test
+  /// cases when they are run, if any.
+  ///
+  /// This property allows optionally filtering the test cases of a
+  /// parameterized test function. The value of this property is invoked once
+  /// for each test function which is run, and if the result is non-`nil`, the
+  /// resulting test case filter function is then invoked once for each of the
+  /// test's associated test cases to determine if they should run. If the
+  /// result of invoking this property is `nil`, then all of the test cases of
+  /// the test will be run (i.e. no filter will be applied).
+  ///
+  /// ## See Also
+  ///
+  /// - ``testFilter-swift.property``
+  /// - ``setTestFilter(toMatch:)``
+  @_spi(ExperimentalParameterizedTesting)
+  public var testCaseFilter: @Sendable (_ test: Test) -> TestCaseFilter? = { _ in nil }
```

Comment:
I'm still confused about this part.

Reply:
I structured it this way for efficiency: For the common case where a particular test does not wish to filter to only run _specific_ test cases, `Runner` will not need to repeatedly invoke the same filter closure for each of that test's cases. It can instead invoke this outer `testCaseFilter` once for each `Test`, and then only if it returned a non-`nil` `TestCaseFilter` closure does it need to repeatedly invoke the returned closure to decide whether each test _case_ should run.

In other words it allows a client to more efficiently indicate, "Do not perform any filtering of the test cases of [TestX]".

Reply:
Seems like that can be accomplished by just passing both the test and test case into the closure, no? Let the caller worry about the nuances like that since they'll be caller-specific.

Reply:
Ok I think this is acceptable, since in the circumstances where this is used, we will typically _also_ be filtering to only run the specific tests which contain those cases